info/v1/container.go

```diff
@@ -530,6 +555,9 @@ type ContainerStats struct {
 	// Task load stats
 	TaskStats LoadStats `json:"task_stats,omitempty"`
 
+	// Metrics for NVIDIA GPUs. Each GPU corresponds to one element in the array.
+	NvidiaGpu []NvidiaGpuStats `json:"nvidia_gpu,omitempty"`
```

Comment:
This assumes that one GPU is used by only one container, right?
Does that mean you won't be able to see the status of GPUs that are not being used by any container? It might still be useful (e.g., they might be overheating, or be in a unhealthy state).

My feeling is that the current `NvidiaGpuStats` struct should be per-node (for all GPUs) and not per-container.
Per-container, you could still try to detect the assigned GPUs, or even better: determine what each container is really using. This could be done by using the NVML API to list the PIDs that have an active context on the GPU and then finding which container these PIDs belong to. This would be similar to what you see when you do `nvidia-smi`.